mmcv/ops/focal_loss.py

```diff
@@ -43,6 +43,11 @@ def forward(ctx,
 
         output = input.new_zeros(input.size())
 
+        if IS_DIPU_AVAILABLE:
```

Comment:
can we put the type cast inside .cpp? Python api better be device agnostic

Reply:
this arises from some bug in `dipu` and `diopi` which is fixed now. will delete these codes.